README.md

```diff
@@ -0,0 +1,1743 @@
+<h1>OpAMP: Open Agent Management Protocol</h1>
+
+
+Author: Tigran Najaryan, Splunk
+
+Contributors: Chris Green, Splunk
+
+Copyright 2021 Splunk Inc., Apache License, Version 2.0
+
+Status: Early Draft (see Open Questions).
+
+September 2021
+
+Note: address all TODO and Open Questions before considering the document ready for final review.
+
+Note 2: this document requires a simplification pass to reduce the scope, size and complexity.
+
+
+[TOC]
+
+
+<h1 id="introduction">Introduction</h1>
+
+
+Open Agent Management Protocol (OpAMP) is a network protocol for remote management of large fleets of data collection Agents.
+
+OpAMP allows Agents to report their status to and receive configuration from a Server and to receive addons and agent installation package updates from the server. The protocol is vendor-agnostic, so the Server can remotely monitor and manage a fleet of different Agents that implement OpAMP, including a fleet of mixed agents from different vendors.
+
+OpAMP supports the following functionality:
+
+
+
+* Remote configuration of the agents.
+* Status reporting. The protocol allows the agent to report the properties of the agent such as its type and version or the operating system type and version it runs on. The status reporting also allows the management server to tailor the remote configuration to individual agents or types of agents.
+* Agent's own telemetry reporting to an [OTLP](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/protocol/otlp.md)-compatible backend to monitor agent's process metrics such as CPU or RAM usage, as well as agent-specific metrics such as rate of data processing.
+* Management of downloadable agent-specific addons.
+* Secure auto-updating capabilities (both upgrading and downgrading of the agents).
+* Connection credentials management, including client-side TLS certificate revocation and rotation.
+
+The functionality listed above enables a 'single pane of glass' management view of a large fleet of mixed agents (e.g. OpenTelemetry Collector, Fluentd, etc).
+
+<h1 id="communication-model">Communication Model</h1>
+
+
+The OpAMP Server manages Agents that implement the client-side of OpAMP protocol. The Agents can optionally send their own telemetry to an OTLP destination when directed so by the OpAMP Server. The Agents likely also connect to other destinations, where they send the data they collect:
+
+
+```
+        ┌────────────┬────────┐           ┌─────────┐
+        │            │ OpAMP  │  OpAMP    │ OpAMP   │
```

Comment:
This is less than a nit but perhaps to emphasise the protocol used and for consistency we could use something along following
```suggestion
        │            │ OpAMP  │  OpAMP/WS │ OpAMP   │
```